ConfigParser has incompatible type IO[Any]; expected TextIO

Hello,

This is a reference from python/mypy#1688.

I am testing mypy and cannot solve this error. I am using Python 3.5.1 and mypy 0.4.2 (same error with mypy 0.4.1). The snippet:

``` python
    #!/usr/bin/env python
    # coding: utf-8

    from configparser import ConfigParser


    def update_stats(ini_file: str, lines: int) -> None:
        ''' Statistics update. '''

        ini = ConfigParser()
        ini.read(ini_file)
        lines += ini.getint('stats', 'lines')
        ini.set('stats', 'lines', str(lines))
        ini.write(open(ini_file, 'w'))

        # Same error with io
        import io
        ini.write(io.open(ini_file, 'w'))


    update_stats('stats.ini', 10)
```

And `stats.ini`:

``` ini
    [stats]
    lines = 2
```

Using the command line:

``` shell
    $ mypy tests.py
    test.py: note: In function "update_stats":
    test.py:14: error: Argument 1 to "write" of "ConfigParser" has incompatible type IO[Any]; expected "TextIO"
    test.py:18: error: Argument 1 to "write" of "ConfigParser" has incompatible type IO[Any]; expected "TextIO"
```

Could you please tell me what is going wrong  here? Thank you in advance :)

--- a/stdlib/3/configparser.pyi
+++ b/stdlib/3/configparser.pyi
@@ -4,7 +4,7 @@
 # reading configparser.py.
 
 from typing import (MutableMapping, Mapping, Dict, Sequence, List,
-                    Iterable, Iterator, Callable, Any, TextIO)
+                    Iterable, Iterator, Callable, Any, IO)
 # Types only used in type comments only
 from typing import Optional, Tuple  # noqa
 
@@ -103,7 +103,7 @@ class ConfigParser(_parser):
     def set(self, section: str, option: str, value: str) -> None: ...
 
     def write(self,
-              fileobject: TextIO,
+              fileobject: IO[str],
               space_around_delimiters: bool = True) -> None: ...
 
     def remove_option(self, section: str, option: str) -> bool: ...

